kin/monitors.py

```diff
@@ -1,227 +1,68 @@
-            for event in self.sse_client:
-                if stop_event.is_set():
-                    return
-                if event.event != 'message':
-                    continue
-                try:
-                    tx = json.loads(event.data)
-
-                    try:
-                        tx_data = SimplifiedTransaction(RawTransaction(tx))
-                    except CantSimplifyError:
-                        continue
-
-                    if tx_data.operation.type != OperationTypes.PAYMENT:
-                        continue
-
-                    if tx_data.source in self.addresses:
-                        self.callback_fn(tx_data.source, tx_data, self)
-                    if tx_data.operation.destination in self.addresses:
-                        self.callback_fn(tx_data.operation.destination, tx_data, self)
-
-                except Exception as ex:
-                    logger.exception(ex)
-                    continue
-        except TypeError:
-            # If we got a type error, that means retry was none, so we should end the thread
-            return
-
-    def stop(self):
-        """
-        Stop monitoring the account.
-
-        The thread will terminate in up to X seconds, where X is the timeout set by the blockchain.
-        """
-
-        # Set the stop event, this will terminate the thread once we get the next event from the blockchain.
-        self.stop_event.set()
-
-        # Change the retry value,
-        # this will cause an exception when trying to reconnect after timeout by the blockchain,
-        # which will terminate the thread.
-        self.sse_client.retry = None
-
-    def add_address(self, address):
-        """
-        Add address to the watched addresses list
-        :param address: address to add
-        """
-        if address in self.addresses:
-            return
-
-        if self.stop_event.is_set():
-            raise StoppedMonitorError()
-
-        if not is_valid_address(address):
-            raise StellarAddressInvalidError('invalid address: {}'.format(address))
-
-        if not self.kin_client.does_account_exists(address):
-            raise AccountNotFoundError(address)
-
-        self.addresses.append(address)
-
-    def remove_address(self, address):
-        """
-        Remove an address for the list of addresses to watch
-        :param address: the address to remove
-        """
-        if self.stop_event.is_set():
-            raise StoppedMonitorError()
-
-        self.addresses.remove(address)
+            tx_data = SimplifiedTransaction(RawTransaction(tx))
+        except CantSimplifyError:
+            logger.debug("SSE transaction couldn't be simplified: ", tx)
+            continue
+
+        if tx_data.operation.type != OperationTypes.PAYMENT:
+            logger.debug("Non-payment SSE transaction skipped: ", tx_data)
+            continue
+
+        # Will yield twice if both of these are correct. (someone sent to himself) - which it fine
```

Comment:
why not check both and send an only single event? (not crucial as it's an edge case, but don't see a reason to fire twice)

Reply:
This is just how it worked so far so I didnt wanted to change it